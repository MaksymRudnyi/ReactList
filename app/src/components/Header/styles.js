import styled from 'styled-components';

const Container = styled.header`
  background-color: #282c34;
  color: white;
`;
Container.displayName = 'Container';

const Content = styled.div`
  max-width: 1024px;
  margin: 0 auto;
  padding: 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;
Content.displayName = 'Content';

export {
  Container,
  Content,
};
