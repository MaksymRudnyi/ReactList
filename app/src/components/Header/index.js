import React from 'react';
import PropTypes from 'prop-types';

import Filter from '../../components/Filter';
import Search from '../../components/Search';
import { Container, Content } from './styles';

const Header = ({searchTerm, specialities, onChange, onFilterChange}) => (
	<Container>
		<Content>
			<Search
				term={searchTerm}
				onChange={onChange}/>

			<Filter
				specialities={specialities}
				onFilterChange={onFilterChange}
			/>
		</Content>
	</Container>
);

export default Header

Header.propTypes = {
	searchTerm: PropTypes.string,
	specialities: PropTypes.objectOf(PropTypes.bool),
	onFilterChange: PropTypes.func.isRequired,
	onChange: PropTypes.func.isRequired,
};
