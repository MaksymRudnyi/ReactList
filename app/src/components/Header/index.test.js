import React from 'react';
import toJson from 'enzyme-to-json';

import { shallow } from 'enzyme';

import Header from './';

describe('Item', () => {
	let sut;
	let props = {
		specialities: ['Excavations', 'Plumbing', 'Electrical'],
		onFilterChange: jest.fn(),
		onChange: jest.fn(),
		searchTerm: '',
	};

	it('should match snapshot', () => {
		sut = shallow(<Header {...props} />);

		expect(toJson(sut)).toMatchSnapshot();
	});
});
