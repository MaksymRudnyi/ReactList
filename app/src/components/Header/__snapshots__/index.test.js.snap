// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match snapshot 1`] = `
<Container>
  <Content>
    <Search
      onChange={[MockFunction]}
      term=""
    />
    <Filter
      onFilterChange={[MockFunction]}
      specialities={
        Array [
          "Excavations",
          "Plumbing",
          "Electrical",
        ]
      }
    />
  </Content>
</Container>
`;
